examples/07 Code Generation.ipynb

```diff
@@ -295,99 +285,90 @@
+    "    for counter, prop in enumerate(cls.properties, start=1):\n",
+    "        multiplicity = (\"\", \"[]\")[prop.max_card.value > 1]\n",
     "        if (\n",
     "            isinstance(\n",
     "                prop.type, capellambse.model.crosslayer.information.Class\n",
     "            )\n",
     "            and prop.type not in current_classes\n",
     "        ):\n",
-    "            current_classes.append([prop.type])\n",
+    "            current_classes.append(prop.type)\n",
     "            nested_text = class_to_proto(\n",
-    "                prop.type, current_classes, indent + \"\\t\"\n",
+    "                prop.type, current_classes, indent + \" \"*4\n",
     "            )\n",
-    "            text.append(nested_text)\n",
-    "            text.append(\n",
-    "                f\"{indent}repeated {prop.type.name}{multiplicity} {prop.name} = {counter+1};\"\n",
+    "            lines.append(nested_text)\n",
+    "            lines.append(\n",
+    "                f\"{indent}repeated {prop.type.name}{multiplicity} {prop.name} = {counter};\"\n",
     "            )\n",
     "        else:\n",
-    "            text.append(\n",
-    "                f\"{indent}{prop.type.name}{multiplicity} {prop.name} = {counter+1};\"\n",
+    "            lines.append(\n",
+    "                f\"{indent}{prop.type.name}{multiplicity} {prop.name} = {counter};\"\n",
     "            )\n",
-    "    text[-1] = text[-1] + f\"\\n{indent[:-1]}}}\"\n",
-    "    text = \"\\n\".join(text)\n",
+    "    lines[-1] = lines[-1] + f\"\\n{indent[:-1]}}}\"\n",
+    "    text = \"\\n\".join(lines)\n",
     "    return text"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The protobuf interface of class _Trajectory_ would look as follows:\n"
+    "The protobuf interface of class `Trajectory` would look as follows:\n"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 10,
+   "execution_count": 29,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
       "# file: Trajectory.proto \n",
-      "message  Trajectory {\n",
-      "\tmessage  Waypoint {\n",
-      "\t\tfloat lat = 1;\n",
-      "\t\tfloat lon = 2;\n",
-      "\t\tfloat alt = 3;\n",
-      "\t\tmessage  Example {\n",
-      "\t\t\tstr test = 1;\n",
-      "\t\t}\n",
-      "\t\trepeated Example[] examples = 4;\n",
-      "\t}\n",
-      "\trepeated Waypoint[] waypoints = 1;\n",
-      "}\n",
+      "   message  Trajectory {\n",
+      "       message  Waypoint {\n",
+      "        float lat = 1;\n",
+      "        float lon = 2;\n",
+      "        float alt = 3;\n",
+      "           message  Example {\n",
+      "            str test = 1;\n",
+      "           }\n",
+      "        repeated Example[] examples = 4;\n",
+      "       }\n",
+      "    repeated Waypoint[] waypoints = 1;\n",
+      "   }\n",
```

Comment:
For some reason, everything is indent with 3 spaces now (and 4 additional spaces per sub-indent). That looks very odd, and might be a bug from a partial refactor. What do you say?